python/mlx/optimizers.py

```diff
@@ -140,6 +140,144 @@ def apply_single(
         return parameter - self.learning_rate * update
 
 
+class RMSprop(Optimizer):
+    r"""Implementation of the RMSprop optimizer [1].
+
+    .. math::
+
+        v_{t+1} &= \beta v_t + (1 - \beta) g_t^2 \\
+        w_{t+1} &= w_t - \lambda \frac{g_t}{\sqrt{v_{t+1} + \epsilon}}
+
+    [1]: Tieleman, T. and Hinton, G. 2012. Lecture 6.5-rmsprop, coursera: Neural networks for machine learning
+    """
+
+    def __init__(self, learning_rate: float, beta: float = 0.99, eps: float = 1e-8):
+        super().__init__()
+
+        self.learning_rate = learning_rate
+        self.beta = beta
+        self.eps = eps
+
+        if self.beta < 0.0:
+            raise ValueError(
+                f"RMSprop beta should be >=0, {self.beta} was provided instead"
+            )
+        if self.eps < 0.0:
+            raise ValueError(
+                f"RMSprop epsilon should be >0, {self.eps} was provided instead"
+            )
+
+    def apply_single(
+        self, gradient: mx.array, parameter: mx.array, state: OptimizerState
+    ):
+        """Performs the RMSprop parameter update and stores :math:`v` in the optimizer state."""
+        lr = self.learning_rate
+        beta = self.beta
+        eps = self.eps
+
+        v = state.get("v", mx.zeros_like(gradient))
+        v = beta * v + (1 - beta) * mx.square(gradient)
+        state["v"] = v
+
+        return parameter - lr * gradient / (mx.sqrt(v) + eps)
+
+
+class Adagrad(Optimizer):
+    r"""Implementation of the Adagrad optimizer [1].
+
+    Our Adagrad implementation follows the original paper. In detail,
+
+    .. math::
+
+        v_{t+1} &= v_t + g_t^2 \\
+        w_{t+1} &= w_t - \lambda \frac{g_t}{\sqrt{v_{t+1} + \epsilon}}
```

Comment:
Please include the `Args` section (as in the SGD example) and you can mention that `\lambda` is the learning rate here.

Same comment for the other optimizers.

Reply:
I guess pre-existing code like adam, adamw also didn't have explanation aboute the \lambda.

In torch and tensorflow optimizers code, they usually do not have detailed explanation, so I did like it.

Do you want to include Args section for all other pre-existing functions as well?

Reply:
I won't ask you to add it for all of them but at least moving forward for the new ones. If you are up for adding the Args section for others that would also be really great!